gleam/src/gilded_rose.gleam

```diff
@@ -0,0 +1,173 @@
+                              False -> 0
+                            }
+                          False -> 0
+                        }
+                      False -> 0
+                    }
+                  }
+                  False -> 0
+                }
+              False -> 0
+            }
+          False -> item.quality
+        }
+      }
+    }
+
+    let new_sell_in = case item.name != "Sulfuras, Hand of Ragnaros" {
+      True -> item.sell_in - 1
+      False -> item.sell_in
+    }
+
+    case new_sell_in < 0 {
+      True ->
+        case item.name != "Aged Brie" {
+          True ->
+            case item.name != "Backstage passes to a TAFKAL80ETC concert" {
+              True ->
+                case new_quality > 0 {
+                  True ->
+                    case item.name != "Sulfuras, Hand of Ragnaros" {
+                      True ->
+                        Item(
+                          ..item,
+                          sell_in: new_sell_in,
+                          quality: new_quality - 1,
+                        )
+                      False -> Item(..item, sell_in: new_sell_in, quality: 80)
+                    }
+                  False ->
+                    Item(..item, sell_in: new_sell_in, quality: new_quality)
+                }
+              False ->
+                Item(
+                  ..item,
+                  sell_in: new_sell_in,
+                  quality: new_quality - new_quality,
+                )
+            }
+          False ->
+            case new_quality < 50 {
+              True ->
+                Item(..item, sell_in: new_sell_in, quality: new_quality + 1)
+              False -> Item(..item, sell_in: new_sell_in, quality: 50)
+            }
+        }
+      False ->
+        case item.name == "Sulfuras, Hand of Ragnaros" {
+          True -> Item(..item, sell_in: new_sell_in, quality: 80)
+          False ->
+            case new_quality > 50 {
+              True -> Item(..item, sell_in: new_sell_in, quality: 50)
+              False -> Item(..item, sell_in: new_sell_in, quality: new_quality)
+            }
+        }
+    }
+  }
+  list.map(inventory, update_quality_item)
+}
+
+pub fn main() {
+  let initial_inventory = [
+    Item("+5 Dexterity Vest", 10, 20),
+    Item("Aged Brie", 2, 0),
+    Item("Elixir of the Mongoose", 5, 7),
+    Item("Sulfuras, Hand of Ragnaros", 0, 80),
+    Item("Sulfuras, Hand of Ragnaros", -1, 80),
+    Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
+    Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
+    Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
+    Item("Conjured Mana Cake", 3, 6),
+  ]
+
+  let days_flag =
+    flag.int()
+    |> flag.default(30)
+    |> flag.description("Number of days")
+
+  let flag = "days"
+
+  let simulate_inventory = fn() {
+    use input <- glint.command()
+    let assert Ok(number_of_days) = flag.get_int(from: input.flags, for: flag)
```

Comment:
Note: I noticed other examples provided an option to select a different number of days. I don't exactly know why and this does bring in more dependencies. If `gleam run` should only produce 30 days of output I can strip that all out.

Reply:
It would drop both `glint` and `argv`

Reply:
Option for days is required but default is not 30 but 2 in most languages. Please change that. Also can we split the main code to file named "Program" or "TextTestFixture" please

Reply:
I made an assumption that you preferred squashed commits. Hopefully that was accurate. I set the default to 2 here: https://github.com/emilybache/GildedRose-Refactoring-Kata/pull/548/commits/c9f452655476985f795a7d9a942bc2ecd32d7f70#diff-506be539568729aa4eb6af30a87b856b47ac987a0b7e48f7e428faf15ab81339R13

The structure is now

-  `src/cli.gleam` which contains the cli application
- `src/gilded_rose.gleam` contains my main function that runs the cli app
- `src/gilded_rose_item.gleam` contains the gilded rose item and update function
- `src/test_fixture.gleam` contains the test fixture and simulation function

Reply:
Thanks for the review!